contracts/src/tokens/drop/DropFacet.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+// interfaces
+import {IDropFacet} from "contracts/src/tokens/drop/IDropFacet.sol";
+// libraries
+import {DropStorage} from "contracts/src/tokens/drop/DropStorage.sol";
+import {CurrencyTransfer} from "contracts/src/utils/libraries/CurrencyTransfer.sol";
+import {BasisPoints} from "contracts/src/utils/libraries/BasisPoints.sol";
+
+// contracts
+import {Facet} from "contracts/src/diamond/facets/Facet.sol";
+import {DropFacetBase} from "contracts/src/tokens/drop/DropFacetBase.sol";
+import {OwnableBase} from "contracts/src/diamond/facets/ownable/OwnableBase.sol";
+
+contract DropFacet is IDropFacet, DropFacetBase, OwnableBase, Facet {
+  using DropStorage for DropStorage.Layout;
+
+  function __DropFacet_init() external onlyInitializing {
+    _addInterface(type(IDropFacet).interfaceId);
+  }
+
+  function claimWithPenalty(
+    uint256 conditionId,
+    address account,
+    uint256 quantity,
+    bytes32[] calldata allowlistProof
+  ) external {
+    DropStorage.Layout storage ds = DropStorage.layout();
+
+    _verifyClaim(ds, conditionId, account, quantity, allowlistProof);
+
+    ClaimCondition storage condition = ds.getClaimConditionById(conditionId);
+
+    uint256 amount = quantity;
+    uint256 penaltyBps = condition.penaltyBps;
+    if (penaltyBps > 0) {
+      uint256 penaltyAmount = BasisPoints.calculate(quantity, penaltyBps);
+      amount = quantity - penaltyAmount;
+    }
+
+    _updateClaim(ds, conditionId, account, amount);
+
+    CurrencyTransfer.safeTransferERC20(
+      condition.currency,
+      address(this),
+      account,
+      amount
+    );
+
+    emit DropFacet_Claimed_WithPenalty(
+      conditionId,
+      msg.sender,
+      account,
+      amount
+    );
+  }
+
+  function claimAndStake(
```

Comment:
What is the dependency needed for this 2nd claim function to be implemented ?

Reply:
#878